hack/build.sh

```diff
@@ -2,6 +2,16 @@
 
 set -ex
 
+version() { echo "$@" | gawk -F. '{ printf("%03d%03d%03d\n", $1,$2,$3); }'; }
```

Comment:
are we sure that `gawk` will be available on any build host? Can we use built-in shell functions?

Reply:
for example there is `printf` command in POSIX as well POSIX also defines the `IFS` variable that can be used to split your parameter into the desired character. Can be done like:
```
version() { printf "%03d%03d%03d\n" "$1" "$2" "$3" }
IFS="."
version $myversion
unset IFS
```
In case `gawk` is part of minimal installation of all OSes we support - Fedora, CentOS, RHEL, then using `gawk` is fine with me.

Reply:
Let's stick to POSIX here. @droslean can you update this PR?